src/ragger/conftest/configuration.py

```diff
@@ -0,0 +1,23 @@
+from typing import Dict, Any
+
+# The following parameters, if any, are required.
+# Please set their values in your 'conftest.py' file
+REQUIRED: Dict[str, Any] = {}
+
+# The following parameters are optional and can be used to customize your tests
+# You can set their values in your 'conftest.py' file if you want to use them
+OPTIONAL: Dict[str, Any] = {
```

Comment:
I'm not keen on using a dictionary at this level. By construction, a dictionary does not hold specific values, just whatever keys, and attached values. Add to this that the dict is called `OPTIONAL`, and I could think as a dev as all these keys/values are totally optional and I could remove them.

I would be  wrong of course, as the conftest file accesses variables such as `conf.OPTIONAL["BACKEND_SCOPE"]` which would fail if `BACKEND_SCOPE` is not defined, or makes assumption on what kind of variable it uses (`conf.OPTIONAL["SIDELOADED_APPS"].items()` means `conf.OPTIONAL["SIDELOADED_APPS"]` is a dict).

Failure is not an issue in itself, but in this case it should happen when the conf is initialized (when `OPTIONAL` is created), rather than when it is used (when the fixture resolves `conf.OPTIONAL[<whatever>]`). This can't be done with a dictionary.

You know I'm a bit biaised toward `dataclasses.dataclas` classes, and I think it would be quite fitting in this case.

```python

@dataclass
class OptionalOptions:
    SIDELOADED_APPS: dict = dict()
    SIDELOADED_APPS_DIR: str = str()
    BACKEND_SCOPE: str = "class"

OPTIONAL = OptionalOptions()
```
... or rather, if you want to explicit to the user where it should be changed:

```python
OPTIONAL = OptionalOptions(
    # use this parameter if you want Speculos to emulate other applications being installed on the device
    # example: {"bitcoin": "Bitcoin", "ethereum": "Ethereum"}
    SIDELOADED_APPS = dict(),
    # Mandatory in case SIDELOADED_APPS is used
    SIDELOADED_APPS_DIR = str(),
    # As the backend instantiation may take some time, Ragger supports multiple backend scopes.
    # You can choose to share the backend instance between {session / module / class / function}
    # When using "session" all your tests will share a single backend instance (faster)
    # When using "function" each test will have its independent backend instance (no collusion)
    BACKEND_SCOPE = "class"
)
```
(and you hide the `class OptionalOptions` declaration in another module).

This way you also get rid of all these "SIDELOADED_APPS" strings buried in the code, which are a pain to maintain, contrary to explicit class properties which will be properly detected/retrieved/corrected by most IDEs.

Reply:
Done with option 2 but I did not hide it in another module, I'm not sure it would be useful: the configuration module provides the class type and it's default value
Option 1 did not work I had issues with `dict()` being refused for default assignation or whatever

Reply:
Great, thanks!